mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -6255,10 +6295,70 @@ struct ThreadwiseCopyRewritePattern
         else
           destLowerIndices = destUpperIndices;
 
+        bool toEmitOOBStoreCheckLogic = false;
+        SmallVector<unsigned, 2> oobStoreCheckDims;
+        if (destLowerIndices.size() == 5) {
+          for (unsigned iter = 0; iter < destTransform.getNumResults();
+               ++iter) {
+            auto expr = destTransform.getResult(iter);
+            if (iter == 2) { // put gemmM or gemmN dim
+              toEmitOOBStoreCheckLogic = true;
+              oobStoreCheckDims.push_back(iter);
+            }
+          }
+        }
+
         // Store to dest.
         // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        if (toEmitOOBStoreCheckLogic) {
+          Value withinStoreBoundsOp =
+              b.create<ConstantIntOp>(loc, 1, b.getIntegerType(1));
+
+          auto iv0 = b.create<IndexCastOp>(loc, destLowerIndices[0],
```

Comment:
IV has its own meaning, induction variable, in a compiler. please rename these variables.